ssl/ssl_rsa.c

```diff
@@ -153,8 +153,11 @@ static int ssl_set_pkey(CERT *c, EVP_PKEY *pkey)
          * Don't check the public/private key, this is mostly for smart
          * cards.
          */
-        if (EVP_PKEY_id(pkey) == EVP_PKEY_RSA
-            && RSA_flags(EVP_PKEY_get0_RSA(pkey)) & RSA_METHOD_FLAG_NO_CHECK) ;
+        if ((EVP_PKEY_id(pkey) == EVP_PKEY_RSA &&
+             RSA_flags(EVP_PKEY_get0_RSA(pkey)) & RSA_METHOD_FLAG_NO_CHECK) ||
+            (EVP_PKEY_id(pkey) == EVP_PKEY_EC &&
+             EC_KEY_get_flags(EVP_PKEY_get0_EC_KEY(pkey)) & EC_FLAG_EXT_PKEY))
```

Comment:
It doesn't seem appropriate for this to be inside the `OPENSSL_NO_RSA` guard - and similarly probably should be in an `OPENSSL_NO_EC` guard.